### PR TITLE
Active user count

### PR DESCRIPTION
**Pull request by Valioesi**
Resolves: https://expedition-grundeinkommen.monday.com/boards/860992020/pulses/1935597190

The pr consists of 3 parts:
- endpoint, which will serve as a callback for mailjet and will update the user's attribute emailActivity, after user has opened or clicked on a mailjet email (after merging the trigger will need to be set in the mailjet gui)
- cron job which analyses db and cognito to generate count and save it as json in s3
- endpoint to retrieve count

Count is generated by: 
- people who have opened email in last 3 months
- people who have been active on website during last 3 months (checking for attribute in user's store)
- because the last attribute will only be set after frontend pr has been merged we check for last sign in by using timestamp of login code which was saved in cognito

Timespan of 3 months can be changed if necessary. I left out donations for now, since using db data is not complete cause it's missing paypal and transfers. Would like to talk about that soon, since I am not sure anyway if we agreed that I should leave it out for now. 

Check if code, tests and doc seem fine. There is no automatic tests for generation of count cause it's not that easy to test. 
Endpoint is here: https://2j0bcp5tr9.execute-api.eu-central-1.amazonaws.com/dev/analytics/active-users

### CODE REVIEW

**Comment by Valioesi** [comment]
Spoke to Tatiana and Nico, leaving out donations for now is fine.

File: serverless/src/shared/users/index.js [hunk 1]
```diff
@@ -228,6 +228,26 @@ const createLoginCode = async ({ email, userId, userPoolIdOverwrite }) => {
   return secretLoginCode;
 };
 
+const getAllCognitoUsers = async (
+  cognitoUsers = [],
+  paginationToken = null
```
Comment (nmauersberg) [inline serverless/src/shared/users/index.js:233]: What is this paginationToken used for?

**Review by nmauersberg — APPROVED** [review]
Cool, I went through the code and it looks very clean. Just left one question about a paginationToken, which I did not understand.

Thank you!

File: serverless/src/shared/users/index.js (hunk 1 above)
Comment (Valioesi) [reply serverless/src/shared/users/index.js:233]: The AWS services (Cognito but dynamo as well) work with pagination, meaning: to get all users I wrote a recursive function which makes multiple requests (always starting on the next "page") and its users to an array each time.

Comment (nmauersberg) [reply serverless/src/shared/users/index.js:233]: Cool, thanks! :)

**Pull request merged**